src/main/resources/templates/semester.html

```diff
@@ -0,0 +1,43 @@
+<!DOCTYPE html>
+<html xmlns:th="http://www.thymeleaf.org"
+      xmlns:layout="http://www.ultraw.net.nz/thymeleaf/layout"
+      layout:decorator="layout">
+<head>
+    <meta charset="UTF-8">
```

Comment:
this is why you are getting a whitelabel error, this line should be
`<meta charset="UTF-8"/>`